pkg/lib/hook/provider.go

```diff
@@ -89,6 +93,27 @@ func (provider *Provider) DispatchEvent(payload event.Payload) (err error) {
 		provider.persistentEventPayloads = append(provider.persistentEventPayloads, payload)
 		err = nil
 
+	case event.BlockingPayload:
+		if provider.Deliverer.WillDeliverBlockingEvent(typedPayload.BlockingEventType()) {
+			seq, err = provider.Store.NextSequenceNumber()
+			if err != nil {
+				err = errorutil.HandledWithMessage(err, "failed to dispatch event")
```

Comment:
HandledWithMessage is for hiding error. If we want to add context to the error, we should write `err = fmt.Errorf("failed to dispatch event: %w", err)`